tests/test_vtk.py

```diff
@@ -45,6 +58,84 @@ def test_pluto_units(pluto_vtk_file):
         assert_allclose_units(getattr(ds, f"{u}_unit"), expected)
 
 
+def test_pluto_complete_units_override(pluto_vtk_file):
+    su = SAMPLE_UNITS
+    unit_combs = distinct_combinations(su, 3)
+    invalid = PlutoVtkDataset.invalid_unit_combinations
+    for comb in unit_combs:
+        uo = {}
+        for unit in comb:
+            uo[unit] = su[unit]
+        if set(uo) not in invalid:
+            ds = yt_idefix.load(
+                pluto_vtk_file["path"], geometry="cartesian", units_override=uo
+            )
+            assert_allclose_units(ds.length_unit, ds.quan(*su["length_unit"]))
+            assert_allclose_units(ds.mass_unit, ds.quan(*su["mass_unit"]))
+            assert_allclose_units(ds.time_unit, ds.quan(*su["time_unit"]))
+            assert_allclose_units(ds.density_unit, ds.quan(*su["density_unit"]))
+            assert_allclose_units(ds.velocity_unit, ds.quan(*su["velocity_unit"]))
+            assert_allclose_units(ds.magnetic_unit, ds.quan(*su["magnetic_unit"]))
+
+
+def test_pluto_one_unit_override(pluto_vtk_file):
+    # Pluto's length_unit and density_unit will be combined with
+    uo = {"time_unit": (2.0, "yr")}
+    ds = yt_idefix.load(pluto_vtk_file["path"], geometry="cartesian", units_override=uo)
+    expect_velocity = ds.length_unit / ds.quan(*uo["time_unit"])
+    assert_allclose_units(ds.velocity_unit, expect_velocity)
+
+
+def test_pluto_two_unit_override(pluto_vtk_file):
+    # Pluto's length_unit will be combined with
+    uo = {"time_unit": (2.0, "yr"), "density_unit": (32.0, "g/cm**3")}
+    ds = yt_idefix.load(pluto_vtk_file["path"], geometry="cartesian", units_override=uo)
+    expect_velocity = ds.length_unit / ds.quan(*uo["time_unit"])
+    expect_mass = ds.quan(*uo["density_unit"]) * ds.length_unit ** 3
+    assert_allclose_units(ds.velocity_unit, expect_velocity)
+    assert_allclose_units(ds.mass_unit, expect_mass)
+
+
+def test_pluto_invalid_units_override(pluto_vtk_file):
+    invalid = PlutoVtkDataset.invalid_unit_combinations
+    for uo in invalid:
+        with pytest.raises(ValueError, match=r".* cannot derive all units\n.*"):
+            yt_idefix.load(
+                pluto_vtk_file["path"], geometry="cartesian", units_override=uo
+            )
+
+
+def test_pluto_temperature_unit_override(pluto_vtk_file):
+    sample = {
+        "time_unit": (2.0, "s"),
+        "length_unit": (4.0, "cm"),
+        "temperature_unit": (2.0, "K"),
+    }
+    with pytest.raises(
+        ValueError,
+        match=(
+            "Temperature is not allowed in units_override, "
+            "since it's always in Kelvin in PLUTO"
+        ),
+    ):
+        yt_idefix.load(
+            pluto_vtk_file["path"], geometry="cartesian", units_override=sample
+        )
+
+
+def test_pluto_over_units_override(pluto_vtk_file):
+    with pytest.raises(
+        ValueError,
+        match=(
+            "More than 3 degrees of freedom were specified "
+            "in units_override \\(6 given\\)"
```

Comment:
please avoid complicated escape sequences (`\\`) and use `re.escape` with raw strings instead.